Source/JavaScriptCore/assembler/MacroAssemblerX86_64.h

```diff
@@ -1247,6 +1247,19 @@ class MacroAssemblerX86_64 : public MacroAssemblerX86Common {
         m_assembler.xchgq_rm(src, dest.offset, dest.base);
     }
 
+    void swapDouble(FPRegisterID reg1, FPRegisterID reg2)
+    {
+        if (reg1 == reg2)
+            return;
+
+        // FIXME: This is kinda a hack since we don't use xmm7 as a temp.
```

Comment:
Also, this is really silly and problematic. Right now, because we do not have callOperation with argumentFPR7, it is working, but we should definitely fix this issue (probably use xmm9 instead and explicitly define xmm9 as fpTempRegister for macro-assembler).